src/components/Trends/index.tsx

```diff
@@ -0,0 +1,67 @@
+import { useEffect, useState } from "react"
+import styled from "styled-components/macro"
+import { useDispatch } from "react-redux"
+import { loadTrends } from "features/trends/trendsSlice"
+import { useAppSelector } from "app/hooks"
+import Pagination from "components/UI/Pagination"
+import TrendPreloader from "components/Trends/TrendPreloader"
+import ErrorText from "components/UI/ErrorText"
+import TrendItem from "./TrendItem"
+
+const Trends = () => {
+	const dispatch = useDispatch()
+	const { data, isLoading, error } = useAppSelector(s => s.trends)
+	const [currentPage, setCurrentPage] = useState(1)
+	const itemPerPage = 5
+	const totalPages = Math.ceil(data.length / itemPerPage)
+	const start = (currentPage - 1) * itemPerPage
+	const end = start + itemPerPage
+
+	useEffect(() => {
+		if (data.length === 0) {
+			dispatch(loadTrends())
+		}
+		window.scrollTo(0, 0)
+	}, [dispatch, currentPage, data.length])
```

Comment:
Если data будет `undefined`, тогда `data.length` вызовет ошибку.

```suggestion
	}, [currentPage, data])
```

Reply:
`data` в redux store по замовчуванню вказано як порожній масив, тому `length `завжди буде. A `dispatch` сам eslint реакта рекомендує вносити